code/code/sys/connect.cc

```diff
@@ -2713,8 +2713,19 @@ void processAllInput()
           continue;
         } 
         if (IS_SET_DELETE(rc, DELETE_VICT)) {
-          delete d->character;
-          d->character = NULL;
+          if (!d->character) continue;
+          
+          if (d == d->character->desc) {
+            vlogf(LOG_BUG, format("d == d->character->desc inside "
+                                  "processAllInput() for %s (Account: %s)") %
+                             d->character->getName() %
+                             (d->account ? d->account->name : "unknown"));
+            delete d->character;
+            d = nullptr;
+          } else {
+            delete d->character;
```

Comment:
```
if (condition) {
  do a
  do b
} else {
  do b
}
```
looks much, much better like:
```
if (condition)
  do a
do b
```

Reply:
I thought about that but wasn't sure how to accomplish it in this case. I need to compare `d` and `d->character->desc` to know whether to set `d` itself to null or only `d->character` to null.

If I factor out the common `delete` call, doing it first, and it turns out `d->character->desc` _was_ pointing to the same desc as `d` then my comparison would cause the same crash I'm trying to fix.

Am I just missing an obvious way around this?

Reply:
Uh, I'm probably missing something - why don't you delete it _afterwards_, unconditionally? You're doing it in both branches already.

Reply:
If `d != d->character->desc` I only want to set `d->character` to null, as `d` should continue to exist but needs to have the pointer to some soon-to-be deleted other object contained in its character property cleared out. 

Whereas if they're equivalent, meaning `d->character` is actually pointing to `d`'s own parent object, I want to set `d` itself to null because the desc `d` is pointing to will be deleted as well as soon as `d->character` is deleted.

I can't call delete until after I've tested for equivalency, as in one case deleting `d->character` will actually delete `d` as well, so how do I factor the delete out while still knowing which needs to be set to null afterwards?

Very possible I'm just confused and/or not aware of something obvious here, though.

Reply:
Aaaaah, so that's what i was missing :D Sorry for the noise.

Reply:
All good, thanks for sanity checking me. 😁 It's important for code like this.